js/common/tags.go

```diff
@@ -0,0 +1,57 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/metrics"
+)
+
+// ApplyCustomUserTags modifies the given metrics.TagSet object with the
+// user specified custom tags.
+// It expects to receive the `keyValues` object in the `{key1: value1, key2:
+// value2, ...}` format.
+func ApplyCustomUserTags(rt *goja.Runtime, tags *metrics.TagSet, keyValues goja.Value) (*metrics.TagSet, error) {
+	if keyValues == nil || goja.IsNull(keyValues) || goja.IsUndefined(keyValues) {
+		return tags, nil
+	}
+
+	keyValuesObj := keyValues.ToObject(rt)
+
+	newTags := tags
+	var err error
+	for _, key := range keyValuesObj.Keys() {
+		if newTags, err = ApplyCustomUserTag(rt, newTags, key, keyValuesObj.Get(key)); err != nil {
+			return nil, err
+		}
+	}
+
+	return newTags, nil
+}
+
+// ApplyCustomUserTag modifies the given metrics.TagSet object with the
+// given custom tag and its value.
+func ApplyCustomUserTag(rt *goja.Runtime, tags *metrics.TagSet, key string, val goja.Value) (*metrics.TagSet, error) {
+	kind := reflect.Invalid
+	if typ := val.ExportType(); typ != nil {
+		kind = typ.Kind()
+	}
+
+	switch kind {
+	case
+		reflect.String,
+		reflect.Bool,
+		reflect.Int64,
+		reflect.Float64:
+
+		return tags.With(key, val.String()), nil
+
+	default:
+		return nil, fmt.Errorf(
+			"setting '%s' metric tag failed;"+
```

Comment:
Why do you want to do that? This will be hit by the users, I think we should keep a user-friendly experience. A code's function name reported in the error I don't helps. I'm happy to improve the message if you think is not clear yet enough.

Reply:
Nope, I didn't mean that we should mention the function name in error what I meant was that the error message ideally should match the function name, which helps maintainers/contributors. 

Like since the error message is saying `"setting '%s' metric tag failed;"` it will be good if the method names something like `SetTag` instead of the applying. Or if the `ApplyTag` will be chosen then the error message should be `"...applying tag failed..."`. So the point is that they should be consistent.

Reply:
Gotcha, I agree! I wait for the final decision about the naming then I will change this in a consistent way.

Reply:
Since users would usually see this error in another context, e.g. when making an http request with `http.get(url, tags: {foo: somethingInvalid})`, I'd suggest something like this:
```suggestion
			"invalid value for metric tag '%s': "+
```